README.md

```diff
@@ -20,16 +20,125 @@
+
+Once a proposal has been created, and the `voting_delay` has elapsed, users can then vote for the proposal (once again, using an `authenticator` as a proxy). 
+
+Voters will need provide the following parameters:
+- `proposal_id`: The ID of the proposal in the space they want to vote in.
+- `voter_address`: The Ethereum address of the proposal creator which will be used to calculate their voting power.
+- `choice`: The votes choice; `FOR`, `AGAINST`, or `ABSTAIN`. 
+-  `voting_strategy_params`: The parameters required by the voting strategies used by the space. 
+
+Once the `voting_duration` has passed, votes are closed, and anyone can call `finalize_proposal` (this time directly on the space contract as no authentication is required): this will finalize the proposal, count the votes for/against/abstain, and call the `executor`. 
+
+Note that each DAO will have at least one space, however a DAO might choose to have multiple spaces if they want to create different 'categories' of proposal each with different settings.
+
+### Voting Strategies
+
+Voting strategies are the contracts used to determine the voting power of a user. Voting strategies can be created in a permissionless way however to use one, one must whitelist the strategy contract in the relevant space contract for the DAO. The most common example is using the ERC-20 token balance of a user to determine his voting power. But we could imagine other voting strategies: owning a specific NFT, owning NFT of collection X and another NFT of collection Y, having participated in protocol xyz... the possibilities are endless! [2] We provide the [single_slot_proof strategy](contracts/starknet/strategies/single_slot_proof.cairo) which allows classic ERC20 and ERC721 balances on L1 (thanks to [Fossil](#Fossil-Storage-Verifier)) to be used as voting power, but feel free to create your own strategies! We hope that the flexibility of the system will unlock a new era of programmable on-chain governance. The interface of a strategy can be found [here](contracts/starknet/strategies/interface.cairo). 
+
+### Fossil Storage Verifier
+
+The backbone of the voting strategies is the Fossil module built by the awesome Oiler team. This module allows any part of the Ethereum mainnet state to be trustlessly verified on StarkNet. Verification of Ethereum state information is achieved by submitting a proof of the state to StarkNet and then verifying that proof. Once this state information has been proved, we can then calculate voting power as an arbitrary function of the information. For more information on Fossil, refer to their [Github](https://github.com/OilerNetwork/fossil)
+
+### Authenticators
+
+Authenticators are the contracts in charge of authenticating users. All authenticators have a function titled `execute` that takes the following arguments: 
+- `to`: The address of the space contract the user wants to interact with. 

```

Comment:
@bonustrack @Orland0x I would prefer to use `target` rather than `to`. I know `to` is used by other contracts (I think Argent), but I find `target` to be much more concise. Wdyt?

Reply:
Maybe we should follow what OZ is doing with Cairo Account Abstraction, but not strong opinion here

Reply:
So OZ uses `to` (see https://github.com/OpenZeppelin/cairo-contracts/blob/dfafdd2ebd778718cf440555e975f6eace278695/src/openzeppelin/account/library.cairo#L26-L29). But it's not very explicit (I believe the `Call` structure is in the library part of the code...).

I find code readability more useful than following "almost standards", and let's say I have a "moderately strong opinion" here haha :p

Reply:
"Can you send 1 eth to the to address" 👎 
"Can you send 1 eth to the target address" 👍  

Much nicer for sure

Reply:
I'm imagining the "drake meme" on this one hahah